jina/serve/runtimes/monitoring.py

```diff
@@ -14,9 +29,208 @@ def _setup_monitoring(self):
+                namespace='jina',
+                labelnames=('runtime_name',),
+                registry=metrics_registry,
+            ).labels(runtime_name)
+
+            self._sent_response_bytes = Summary(
+                'sent_response_bytes',
+                'The size in bytes of the request returned to the client',
+                namespace='jina',
+                labelnames=('runtime_name',),
+                registry=metrics_registry,
+            ).labels(runtime_name)
+
+        else:
+            self._receiving_request_metrics = None
+            self._pending_requests_metrics = None
+            self._failed_requests_metrics = None
+            self._successful_requests_metrics = None
+            self._request_size_metrics = None
+            self._sent_response_bytes = None
+
+        if meter:
+            self._receiving_request_histogram = meter.create_histogram(
+                name='jina_receiving_request_seconds',
+                description='Time spent processing successful request',
+            )
+
+            self._pending_requests_up_down_counter = meter.create_up_down_counter(
+                name='jina_number_of_pending_requests',
+                description='Number of pending requests',
+            )
+
+            self._failed_requests_counter = meter.create_counter(
+                name='jina_failed_requests',
+                description='Number of failed requests',
+            )
+
+            self._successful_requests_counter = meter.create_counter(
+                name='jina_successful_requests',
+                description='Number of successful requests',
+            )
+
+            self._request_size_histogram = meter.create_histogram(
+                name='jina_received_request_bytes',
+                description='The size in bytes of the request returned to the client',
+            )
+
+            self._sent_response_bytes_histogram = meter.create_histogram(
+                name='jina_sent_response_bytes',
+                description='The size in bytes of the request returned to the client',
+            )
+        else:
+            self._receiving_request_histogram = None
+            self._pending_requests_up_down_counter = None
+            self._failed_requests_counter = None
+            self._successful_requests_counter = None
+            self._request_size_histogram = None
+            self._sent_response_bytes_histogram = None
+        self._metric_labels = {'runtime_name': runtime_name}
+
+    def _update_start_request_metrics(self, request: 'Request'):
+        if self._request_size_metrics:
+            self._request_size_metrics.observe(request.nbytes)
+        if self._request_size_histogram:
+            self._request_size_histogram.record(
+                request.nbytes, attributes=self._metric_labels
+            )
+
+        if self._receiving_request_metrics:
+            self._request_init_time[request.request_id] = time.time()
```

Comment:
Not related to this PR but I want to mention it. This map is not always safely cleared after every request ends. Especially in the head `_gather_worker_tasks` is not catching all exceptions. Right now it's focused on the `AioRpcError` and `InternalNetworkError` for retrying. In case if something else happens then the the `_update_end_successful_requests_metrics` method will not correctly remove entries from the dict.

Reply:
let's open a ticket for this, I am not sure there can be other errors,it should be handled in another PR